libcontainer/seccomp/seccomp_linux.go

```diff
@@ -42,7 +42,19 @@ func InitSeccomp(config *configs.Seccomp) (int, error) {
 
 	// Ignore the error since pre-2.4 libseccomp is treated as API level 0.
 	apiLevel, _ := libseccomp.GetAPI()
+	const writeSyscall = "write"
+	writeAllowed := false
 	for _, call := range config.Syscalls {
+		// If write is in the allowed list of syscalls, then we can also notify on the defaultAction.
+		if call.Name == writeSyscall {
+			switch call.Action {
+			case configs.Allow, configs.Trace, configs.Log:
+				writeAllowed = true
+			default:
+				writeAllowed = false
+			}
+		}
```

Comment:
I would move this to another function and then just say here (pseudo code and not sure this function names are good ones, just to clearly express my idea):

`if allowedSyscall("write") && defaultAction is not write ....`

WAIT, no. As I mentioned in the other comment, this is not good enough to know if write is allowed. More info in the other comment :)

But with the other comment and suggestions there in mind, this definitely makes sense to be a different function, maybe stress the fact that we check that the syscall is _always_ allowed (no conditions) and explain this problem in a comment there.